app/furniture/marketplace/current_person.rb

```diff
@@ -0,0 +1,17 @@
+class Marketplace
+  module CurrentPerson
+    def current_person
+      super
+
+      if @current_person.is_a?(Person) || @current_person.is_a?(Guest)
+        return @marketplace_current_person = @current_person
+      end
+
+      @current_person = if @current_person.authenticated?
```

Comment:
This method is confusing to me. In which cases might a @current_person _not_ be a `Person` nor a `Guest`, but still be `authenticated?`

Reply:
Yea, I think there's some... weirdness here. I'll take a quick refactor pass to get rid of some of the stumbling blocks; and we can sit on it since I think the bug is fixed already.

Reply:
That feels like a good call. Hopefully next week I'll have a bit more energy to dig into this.

Reply:
The pass is done; but it doesn't really reduce complexity more than remove cruft.

Anyway; I don't see this as urgent; so we can close it out if you don't get to it int he next week or so.